### PR TITLE
Spring-boot-intro-solution

### PR DESCRIPTION
**Pull request by VArabchuk**

### CODE REVIEW

File: src/main/java/mate/academy/intro/IntroApplication.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    public static void main(String[] args) {
+        SpringApplication.run(IntroApplication.class, args);
+    }
+
+    @Bean
+    public CommandLineRunner commandLineRunner() {
+        return new CommandLineRunner() {
+            @Override
+            public void run(String... args) throws Exception {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/intro/IntroApplication.java]:
```suggestion
        return args -> {
```

File: src/main/java/mate/academy/intro/model/Book.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Data
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/intro/model/Book.java]:
```suggestion
@Getter
@Setter
@Table(name....
```

File: src/main/java/mate/academy/intro/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String title;
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/intro/model/Book.java:24]: Add required constraints

File: src/main/java/mate/academy/intro/repository/BookRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+import java.util.List;
+import mate.academy.intro.model.Book;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/intro/repository/BookRepositoryImpl.java:12]:
```suggestion
@Repository
@RequiredArgsConstructor
```

File: src/main/java/mate/academy/intro/repository/BookRepositoryImpl.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/intro/repository/BookRepositoryImpl.java]:
```suggestion
```

File: src/main/java/mate/academy/intro/service/BookServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
+package mate.academy.intro.service;
+
+import java.util.List;
+import mate.academy.intro.model.Book;
+import mate.academy.intro.repository.BookRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/intro/service/BookServiceImpl.java:9]:
```suggestion
@Service
@RequiredArgsConstructor
```

File: src/main/java/mate/academy/intro/service/BookServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+import java.util.List;
+import mate.academy.intro.model.Book;
+import mate.academy.intro.repository.BookRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BookServiceImpl implements BookService {
+    @Autowired
+    private BookRepository bookRepository;
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/intro/service/BookServiceImpl.java]:
```suggestion
    private final BookRepository bookRepository;
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/mate/academy/intro/model/Book.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+@Entity
+@Getter
+@Setter
+@ToString
+@Table(name = "books")
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(name = "title", nullable = false)
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/intro/model/Book.java]:
```suggestion
    @Column(nullable = false)
```

remove attribute name for all fields

File: src/main/java/mate/academy/intro/repository/BookRepositoryImpl.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(book);
+            transaction.commit();
+            return book;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't save book '" + book + "' to DB", e);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/intro/repository/BookRepositoryImpl.java]:
```suggestion
            throw new DataProcessingException("Can't save book '" + book + "' to DB", e);
```

File: src/main/java/mate/academy/intro/repository/BookRepositoryImpl.java [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+            }
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("FROM Book", Book.class).getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't find all books from DB", e);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/intro/repository/BookRepositoryImpl.java]:
```suggestion
            throw new DataProcessingException("Can't find all books from DB", e);
```

File: src/test/java/mate/academy/intro/IntroApplicationTests.java [hunk 11]
```diff
@@ -0,0 +1,13 @@
+package mate.academy.intro;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
+class IntroApplicationTests {
+
+	@Test
```
Comment (Elena-Bruyako) [inline src/test/java/mate/academy/intro/IntroApplicationTests.java:8]: refactor format (tabulation)

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

**Review by okuzan — APPROVED** [review]

**Pull request merged**